include/beman/optional26/optional.hpp

```diff
@@ -197,16 +197,16 @@ struct hash<optional<T>>;
 
 namespace detail {
 template <class T, class U>
-concept enable_forward_value = std::is_constructible_v<T, U&&> && !std::is_same_v<std::decay_t<U>, in_place_t> &&
-                               !std::is_same_v<optional<T>, std::decay_t<U>>;
+concept enable_forward_value = !std::is_same_v<std::decay_t<U>, optional<T>> &&
+                               !std::is_same_v<std::decay_t<U>, in_place_t> && std::is_constructible_v<T, U&&>;
 
 template <class T, class U, class Other>
 concept enable_from_other =
-    std::is_constructible_v<T, Other> && !std::is_constructible_v<T, optional<U>&> &&
-    !std::is_constructible_v<T, optional<U>&&> && !std::is_constructible_v<T, const optional<U>&> &&
-    !std::is_constructible_v<T, const optional<U>&&> && !std::is_convertible_v<optional<U>&, T> &&
-    !std::is_convertible_v<optional<U>&&, T> && !std::is_convertible_v<const optional<U>&, T> &&
-    !std::is_convertible_v<const optional<U>&&, T>;
+    !std::is_same_v<T, U> && std::is_constructible_v<T, Other> && //
```

Comment:
I don't think the `//` at the end of line needs to be there?

Reply:
Yes, it doesn't need to be there. Sometimes I do this out of habit to force clang-format to break a line. In this case it makes it so that the next two lines are the `is_constructible_v`s and the two lines after that are the "matching" `is_convertible_v`s.

Should I remove it? The diff would certainly look nicer.

Reply:
I think removing it will make the diff look a little cleaner. I'm making a note to come back to this to clean up formatting, or refactor into smaller concepts, so it's more obvious what is intended and how they fit together. 

The primary template was really, originally, just to have a base case for the optional<T&> part and I didn't expect to have to revisit it.

Reply:
OK, I reverted the formatting change and merged main.

Reply:
Would it be OK to put you in the acknowledgements for the next rev of the paper? It's going out in the next few days and hopefully reviewed and forwarded for C++26 in a few weeks.

Reply:
> Would it be OK to put you in the acknowledgements for the next rev of the paper? It's going out in the next few days and hopefully reviewed and forwarded for C++26 in a few weeks.

Sure, thank you!